src/app/controllers/encrypt-submissions.server.controller.js

```diff
@@ -223,88 +225,120 @@ exports.saveResponseToDb = function (req, res, next) {
  */
 exports.getMetadata = function (req, res) {
   let pageSize = 10
-  let { page } = req.query || {}
+  let { page, submissionId } = req.query || {}
   let numToSkip = parseInt(page - 1 || 0) * pageSize
 
-  Submission.aggregate([
-    {
-      $match: {
-        form: req.form._id,
-        submissionType: 'encryptSubmission',
+  let matchClause = {
+    form: req.form._id,
+    submissionType: 'encryptSubmission',
+  }
+
+  if (submissionId) {
+    if (mongoose.Types.ObjectId.isValid(submissionId)) {
+      matchClause._id = mongoose.Types.ObjectId(submissionId)
+      // Reading from primary to avoid any contention issues with bulk queries on secondary servers
+      Submission.findOne(matchClause, { created: 1 })
+        .read('primary')
+        .exec((err, result) => {
+          if (err) {
+            logger.error(getRequestIp(req), req.url, req.headers, err)
```

Comment:
This should follow our standard logging format (`message`, `meta`) to make searching the logs easier